modules/net-vpc/main.tf

```diff
@@ -77,6 +67,58 @@ locals {
     ? try(google_compute_network.network.0, null)
     : try(data.google_compute_network.network.0, null)
   )
+
+  _subnet_data = var.data_folder == null ? {} : {
+    for f in fileset(var.data_folder, "**/*.yaml") :
+    trimsuffix(basename(f), ".yaml") => merge(
+      yamldecode(file("${var.data_folder}/${f}")),
+      {
+        project_id = var.project_id
+        network    = local.network.name
+      }
+    )
+  }
+
+  subnet_data = var.data_folder == null ? {} : {
```

Comment:
you already check for `var.data_folder` in `_subnet_data`

Reply:
Do we really need a second step? The variables are all there in `_subnet_data` already. I would move the `try` in the resource, and do away with most of these locals.

Reply:
Ack - cleaned check in `subnet_data`

I reckon we do need the second step, as yamldecode doesn't allow us to further manipulate datafields. Might be wrong.

Reply:
Resolved in a0f1593